packages/runtime/runtime-utils/src/summaryUtils.ts

```diff
@@ -329,3 +334,29 @@ export function convertSummaryTreeToITree(summaryTree: ISummaryTree): ITree {
         unreferenced: summaryTree.unreferenced,
     };
 }
+
+export class TelemetryContext implements ITelemetryContext {
+    private telemetry = new Map<string, any>();
+
+    /**
+     * {@inheritDoc @fluidframework/runtime-definitions#ITelemetryContext.set}
+     */
+    set(prefix: string, property: string, value: any): void {
+        this.telemetry[`${prefix}${property}`] = value;
+    }
+
+    /**
+     * {@inheritDoc @fluidframework/runtime-definitions#ITelemetryContext.get}
+     */
+    get(prefix: string, property: string): any | undefined {
+        const key = `${prefix}${property}`;
+        return this.telemetry.has(key) ? this.telemetry[key] : undefined;
```

Comment:
I believe you are using Map wrong.
It's a object, so you can do [key], but if you go that route, then I'd rather use {[key: string]: number} type.
If you want to use Map, use get() & set() instead.
Either way, no need to test for undefined - in both cases (using object properties or map) reading non-existing value will return undefined.

Reply:
It seems Map performance is better when we have frequent additions and removals of key value pairs.  I also saw this: 
[https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map#setting_object_properties](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map#setting_object_properties)